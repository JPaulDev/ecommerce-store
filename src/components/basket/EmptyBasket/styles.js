import styled from 'styled-components';

export const Section = styled.section`
  align-items: center;
  background-color: white;
  border: 1px solid var(--color-grey-12);
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 600px;
  padding: 30px;
  text-align: center;

  p {
    margin-bottom: 20px;
  }
`;

export const Heading = styled.h1`
  font-family: 'Uni Sans', sans-serif;
  font-size: var(--font-size-32);
  font-weight: var(--font-weight-semibold);
  margin: 10px 0;
`;
