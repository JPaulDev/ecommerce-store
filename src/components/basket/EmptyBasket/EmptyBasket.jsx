import { Basket } from '../../icons';
import { LinkButton } from '../../ui';
import * as Styled from './styles';

export default function EmptyBasket() {
  return (
    <Styled.Section>
      <Basket width={65} height={57} fill="var(--color-orange-1)" />
      <Styled.Heading>Your Basket</Styled.Heading>
      <p>Your basket is empty, please add some products.</p>
      <LinkButton href="/">
        Continue Shopping
      </LinkButton>
    </Styled.Section>
  );
}
