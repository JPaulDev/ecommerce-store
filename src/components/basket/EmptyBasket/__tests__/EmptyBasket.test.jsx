import renderer from 'react-test-renderer';
import EmptyBasket from '../index';

describe('EmptyBasket component', () => {
  it('renders message that displays when the users basket is empty', () => {
    const tree = renderer.create(<EmptyBasket />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
