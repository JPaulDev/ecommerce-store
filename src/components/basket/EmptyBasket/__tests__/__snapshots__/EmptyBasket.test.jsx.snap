// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyBasket component renders message that displays when the users basket is empty 1`] = `
.c2 {
  background-image: linear-gradient( to bottom, var(--color-grey-7), var(--color-grey-4) );
  border-radius: var(--border-radius-lg);
  box-shadow: var(--box-shadow-md);
  color: white;
  display: inline-block;
  font-size: var(--font-size-13);
  font-weight: var(--font-weight-bold);
  padding: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
}

.c2:hover {
  box-shadow: inset 0 0 0 100px rgb(0 0 0 / 10%),var(--box-shadow-hover);
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  border: 1px solid var(--color-grey-12);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 600px;
  padding: 30px;
  text-align: center;
}

.c0 p {
  margin-bottom: 20px;
}

.c1 {
  font-family: 'Uni Sans',sans-serif;
  font-size: var(--font-size-32);
  font-weight: var(--font-weight-semibold);
  margin: 10px 0;
}

<section
  className="c0"
>
  <svg
    fill="var(--color-orange-1)"
    height={57}
    viewBox="0 0 2048 1792"
    width={65}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1920,768q52.9,0,90.5,37.5T2048,896q0,52.9-37.5,90.5T1920,1024h-15l-115,662q-7.9,46.1-44,76t-82,30H384q-46,0-82-30t-44-76L143,1024H128q-52.9,0-90.5-37.5T0,896q0-52.9,37.5-90.5T128,768ZM485,1568q26-1.9,43.5-22.5c11.7-13.7,16.8-29.2,15.5-46.5l-32-416q-2-25.9-22.5-43.5c-13.7-11.7-29.2-16.8-46.5-15.5s-31.8,8.8-43.5,22.5-16.8,29.2-15.5,46.5l32,416a64.3,64.3,0,0,0,64,59Zm411-64V1088q0-25.9-19-45t-45-19q-25.9,0-45,19t-19,45v416q0,25.9,19,45t45,19q25.9,0,45-19T896,1504Zm384,0V1088q0-25.9-19-45t-45-19q-25.9,0-45,19t-19,45v416q0,25.9,19,45t45,19q25.9,0,45-19T1280,1504Zm352,5,32-416c1.3-17.3-3.8-32.8-15.5-46.5s-26.2-21.2-43.5-22.5-32.8,3.8-46.5,15.5-21.2,26.2-22.5,43.5l-32,416c-1.3,17.3,3.8,32.8,15.5,46.5s26.2,21.2,43.5,22.5h5a64.3,64.3,0,0,0,64-59ZM476,292,383,704H251L352,263q19-88,89-143.5T601,64H768q0-25.9,19-45T832,0h384q25.9,0,45,19t19,45h167q90,0,160,55.5T1696,263l101,441H1665l-93-412q-10.9-44-45.5-72T1447,192H1280q0,26-19,45t-45,19H832q-25.9,0-45-19t-19-45H601q-45,0-79.5,28T476,292Z"
    />
  </svg>
  <h1
    className="c1"
  >
    Your Basket
  </h1>
  <p>
    Your basket is empty, please add some products.
  </p>
  <a
    className="c2"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Continue Shopping
  </a>
</section>
`;
